.github/workflows/tests.yml

```diff
@@ -53,71 +56,224 @@ jobs:
 
       - name: Check upstream
         if: github.event_name == 'pull_request' && matrix.runtime-version == 'latest'
-        run: |
-          COMMIT="$(git log -n 1 --pretty=format:%s HEAD^2)"
-          if [[ "$COMMIT" == *"test-upstream"* ]]
-          then
-            export TEST_UPSTREAM="true"
-            # Put TEST_UPSTREAM into $GITHUB_ENV so it can be used in subsequent workflow steps
-            echo TEST_UPSTREAM=$TEST_UPSTREAM >> $GITHUB_ENV
-            echo "Using development version of dask and distributed"
-          fi
+        run: bash ci/scripts/check_upstream.sh
 
       - name: Install coiled-runtime
+        env:
+          COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
+        run: bash ci/scripts/install_coiled_runtime.sh
+
+      - name: Export environment
+        env:
+          COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        run: bash ci/scripts/export_conda_env.sh
+
+      - name: Build Coiled Software Environment
+        if: ${{ matrix.runtime-version == 'latest' }}
+        env:
+          DASK_COILED__TOKEN: ${{ secrets.COILED_BENCHMARK_BOT_TOKEN }}
+          GITHUB_EVENT_NAME: ${{ github.event_name }}
+          GITHUB_EVENT_NUMBER: ${{ github.event.number }}
+          PYTHON_VERSION: ${{ matrix.python-version }}
+          TEST_NAME: runtime
+        run: bash ci/scripts/build_coiled_env.sh
+
+      - name: Run Coiled Runtime Tests
+        continue-on-error: true
+        id: test
+        env:
+            DASK_COILED__TOKEN: ${{ secrets.COILED_BENCHMARK_BOT_TOKEN }}
+            AWS_ACCESS_KEY_ID: ${{ secrets.RUNTIME_CI_BOT_AWS_ACCESS_KEY_ID }}
+            AWS_SECRET_ACCESS_KEY: ${{ secrets.RUNTIME_CI_BOT_AWS_SECRET_ACCESS_KEY }}
+            COILED_SOFTWARE_NAME: ${{ env.COILED_SOFTWARE_NAME }}
+            COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
         run: |
-          # If testing the latest `coiled-runtime` then install packages defined in `recipe/meta.yaml`
-          # Otherwise, just install directly from the coiled / conda-forge channel
-          if [[ ${{ matrix.runtime-version }} = 'latest' ]]
+          # Ensure we run additional tests when testing the latest coiled-runtime
+          if [[ $COILED_RUNTIME_VERSION = 'latest' ]]
           then
-            python ci/create_latest_runtime_meta.py
-            mamba env update --file latest.yaml
+            export EXTRA_OPTIONS="--run-latest"
           else
-            mamba install -c conda-forge coiled-runtime=${{ matrix.runtime-version }}
+            export EXTRA_OPTIONS=" "
           fi
 
+          python -m pytest $EXTRA_OPTIONS tests/runtime
```

Comment:
It seems like we should move this to a separate script as well

Reply:
done